k8s.gcr.io/Vanity-Domain-Flip.md

```diff
@@ -0,0 +1,191 @@
+  uses GCP resources to check that the backup scripts work as intended in
+  `ci-k8sio-backup`.
+
+## Critical User Journey for Promotion
+
+In order for a user to push to `k8s-artifacts-prod`, they must:
+
+1. Ensure that they have a [subproject staging repo][staging-subproject] (e.g.,
+   `gcr.io/k8s-staging-foo` for the `foo` subproject).
+2. Add the promotion metadata in the [manifests subdirectory][] in the k8s.io repo.
+
+### Security Restrictions
+
+- **Write-once**: Images promoted to production will NOT be deleted, unless under extreme,
+  emergency circumstances that require human supervision (see "Breakglass"
+  section below).
+- **Immutable tags**: New images added to the promoter manifests cannot use an
+  existing tag for the same image. In other words, tags (once created for an
+  image) cannot be deleted.
+- **Mandatory subproject prefix**: Images must be prefixed in production by the
+  name of the subproject. For example, the subproject named `foo` must only push
+  images to `{asia,eu,us}.gcr.io/k8s-artifacts-prod/foo/...`.
+
+# Breakglass
+
+Images in `k8s-artifacts-prod` are not normally deletable. For emergencies,
+however, you can reach the GCR admins listed in the
+`k8s-infra-artifact-admins@kubernetes.io` group [here][groups] who have write
+access to the GCS buckets behind GCR.
+
+# Backups
+
+The GCR images in `k8s-artifacts-prod` are backed up every hour. This is done
+with the `ci-k8sio-backup` job. All images are backed up, even legacy images
+that appeared before the promoter went online that were not tagged and can only
+be referenced by their digest.
+
+## Disaster Recovery
+
+In the event that the `k8s-artifacts-prod` GCR is compromised, a human from the
+`k8s-infra-artifact-admins@kubernetes.io` group must restore from a known-good
+backup snapshot. An example might be:
+
+```
+for region in asia eu us; do
+    gcrane cp -r ${region}.gcr.io/k8s-artifacts-prod-bak/2020/01/01/00 ${region}.gcr.io/k8s-artifacts-prod
+done
+```
+
+# Auditing (cip-auditor)
+
+All GCR stateful changes to `{asia,eu,us}.gcr.io/k8s-artifacts-prod` are
+detected by the auditor, which runs as a service in Cloud Run. If the change
+fits with the intent of the [promoter manifests][k8sio], nothing happens.
+However, if there is a disagreement, then the GCR transaction is marked as
+"REJECTED" and an alert is sent to Stackdriver Error Reporting, where by default
+it currently notifies the project owner via email.
+
+The step-by-step process is:
+
+1. An image is created (new tag), deleted, etc on the `k8s-artifacts-prod` GCR.
+2. Cloud Pub/Sub message with the stateful change contents is sent over HTTPS to the `cip-auditor` service in Cloud Run.
+3. `cip-auditor` clones a fresh copy of [promoter manifests][k8sio] at https://github.com/kubernetes/k8s.io.
+4. `cip-auditor` checks the Pub/Sub message contents against the promoter manifests.
+5. If the message agrees with the promoter manifests, nothing happens.
```

Comment:
Where are logs for this?  Config for cloud run?

Reply:
I need to stand up the cip-auditor service in the first place in the k8s-artifacts-prod project. I can submit a deployment script (as part of this PR) that will do this for me. The script may have issues, but it's better than nothing. Ack